### ISSUE TITLE
Can't install R in Docker

### ISSUE THREAD

**Issue by dragoonis** [openanalytics/shinyproxy#48]
Hey guys, 

Paul from Oasis here.

We're struggling to install R. We have to move from base image `ubuntu:17.04` because it's now EOL and our option is to switch to ubuntu:16.04 LTS, or ubuntu:17.10 - once switchin we're getting apt-get errors and wondering if you can help us out.

```
FROM ubuntu:17.10

ARG USER_ID

RUN apt-get update && \
    apt-get install -y gcc g++ && \
    apt-get install -y --no-install-recommends libcurl4-gnutls-dev libcairo2-dev libxt-dev libssl-dev libssh2-1-dev libssl1.0.0 apt-utils && \
    apt-get install -y --no-install-recommends -f libxml2-dev libssl-dev libiodbc2 libiodbc2-dev && \
    apt-get install -y --no-install-recommends unixodbc unixodbc-dev freetds-dev freetds-bin tdsodbc libgeos-dev && \
    rm -rf /var/lib/apt/lists/*

# todo - remove this dual apt-get update
# Install R
RUN apt-get update && \
    apt-get install -y --no-install-recommends software-properties-common && \

    # todo - put this at the start
    add-apt-repository "deb http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/" && \
    apt-get update && \
    apt-get install -y --no-install-recommends --allow-unauthenticated r-base && \
    rm -rf /var/lib/apt/lists/*

# Install Shiny Server dependencies
# RUN apt-get install -y \
#     gdebi-core pandoc pandoc-citeproc libcurl4-gnutls-dev libcairo2 libxt-dev && \
#     rm -rf /var/lib/apt/lists/*

# Install Flamingo dependencies
RUN R -e "install.packages(c('devtools', 'DT', 'shiny', 'shinyBS', 'httr', 'rjson', 'odbc', 'RODBC', 'leaflet', 'ggplot2', 'curl', 'RCurl', 'xml2', 'XML', 'rmarkdown', 'logging'), repos='https://cloud.r-project.org/')" && \
    R -e "install.packages(c('Dataset'), repos='http://R-Forge.R-project.org')" && \
    R -e "install.packages(c('shinyjs'), repos='https://cloud.r-project.org/')" && \
    R -e "remove.packages(c('devtools'))"

# Create a jenkins user

WORKDIR /var/www/oasis

RUN adduser --shell /bin/bash --uid $USER_ID --disabled-password --gecos "" jenkins && \
    chown -R jenkins:jenkins /var/www/oasis

# Create the logging directory
RUN mkdir /var/log/shiny-server

# Install the flamingo package
COPY flamingo.tar.gz /tmp
RUN R CMD INSTALL /tmp/flamingo.tar.gz

# COPY shiny-server.sh /home/Flamingo/
COPY freetds.conf /etc/freetds/
COPY odbc.ini /etc/
COPY odbcinst.ini /etc/
# COPY shiny-server.conf /etc/shiny-server/
COPY db_conf.sh /home/Flamingo/
COPY Rprofile.site /usr/lib/R/etc/

EXPOSE 3838

USER jenkins

CMD ["R", "-e", "flamingo::runFlamingo()"]

```

Error output:
```

.... SNIP ...

Ign:4 http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/ InRelease

Get:6 http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/ Packages [53.4 kB]

Fetched 57.0 kB in 0s (188 kB/s)

Reading package lists...

W: GPG error: http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 51716619E084DAB9

W: The repository 'http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/ InRelease' is not signed.

Reading package lists...

Building dependency tree...

Reading state information...

Some packages could not be installed. This may mean that you have

requested an impossible situation or if you are using the unstable

distribution that some required packages have not yet been created

or been moved out of Incoming.

The following information may help to resolve the situation:



The following packages have unmet dependencies:

 r-base : Depends: r-base-core (>= 3.4.3-1zesty0) but it is not going to be installed

          Depends: r-recommended (= 3.4.3-1zesty0) but it is not going to be installed

E: Unable to correct problems, you have held broken packages.

Removing intermediate container 693560cea091

The command '/bin/sh -c apt-get update &&     apt-get install -y --no-install-recommends software-properties-common &&     add-apt-repository "deb http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/" &&     apt-get update &&     apt-get install -y --no-install-recommends --allow-unauthenticated r-base &&     rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100

script returned exit code 100
```

**Comment by tverbeke** [comment]
Our own images are all based on Ubuntu LTS 16.04. See e.g. https://github.com/openanalytics/r-base/blob/master/Dockerfile

Closing since not related to ShinyProxy.

**Comment by dragoonis** [comment]
Thanks for showing us your Dockerfile

**Comment by dseynaev** [comment]
Cfr. the trace: 
`
GPG error: http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 51716619E084DAB9
`

You are missing the public key for Ubuntu archives on CRAN.

I would change:
`  add-apt-repository "deb http://cran.ma.imperial.ac.uk/bin/linux/ubuntu zesty/" && \`
to:
`
    add-apt-repository "deb https://cloud.r-project.org/bin/linux/ubuntu artful/" && \
    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E084DAB9 && \
`
